pkg/cloudprovider/vsphereparavirtual/loadbalancer_test.go

```diff
@@ -44,385 +40,310 @@ import (
+	testK8sServiceName      = "test-lb-service"
+	testK8sServiceNameSpace = "test-service-ns"
+	testOwnerReference      = metav1.OwnerReference{
 		APIVersion: "v1alpha1",
 		Kind:       "TanzuKubernetesCluster",
 		Name:       testClustername,
 		UID:        "1bbf49a7-fbce-4502-bb4c-4c3544cacc9e",
 	}
 )
 
-func TestLoadBalancer(t *testing.T) {
-	RegisterFailHandler(Fail)
-	RunSpecs(t, "LoadBalancer Suite")
+func newTestLoadBalancer() (cloudprovider.LoadBalancer, *util.FakeClientWrapper) {
+	scheme := runtime.NewScheme()
+	_ = vmopv1alpha1.AddToScheme(scheme)
+	fc := fakeClient.NewFakeClientWithScheme(scheme)
+	fcw := util.NewFakeClientWrapper(fc)
+	vms := vmservice.NewVMService(fcw, testClusterNameSpace, &testOwnerReference)
+	return &loadBalancer{
+		vmService: vms,
+	}, fcw
 }
 
-var _ = Describe("LoadBalancer Support", func() {
-	var (
-		lb                      cloudprovider.LoadBalancer
-		testK8sService          *v1.Service
-		testK8sServiceName      string
-		testK8sServiceNameSpace string
-
-		// FakeClientWrapper allows functions to be replaced for fault injection
-		fcw *util.FakeClientWrapper
-	)
-
-	newTestLoadBalancer := func() (cloudprovider.LoadBalancer, error) {
-		scheme := runtime.NewScheme()
-		_ = vmopv1alpha1.AddToScheme(scheme)
-		fc := fakeClient.NewFakeClientWithScheme(scheme)
-		fcw = util.NewFakeClientWrapper(fc)
-		vms := vmservice.NewVMService(fcw, testClusterNameSpace, &testOwnerReference)
-		return &loadBalancer{
-			vmService: vms,
-		}, nil
+func TestNewLoadBalancer(t *testing.T) {
+	testCases := []struct {
+		name    string
+		testEnv *envtest.Environment
+		err     error
+	}{
+		{
+			name:    "NewLoadBalancer: when everything is ok",
+			testEnv: &envtest.Environment{},
+			err:     nil,
+		},
 	}
 
-	BeforeEach(func() {
-		lb, _ = newTestLoadBalancer()
-		testK8sServiceName = "test-lb-service"
-		testK8sServiceNameSpace = "test-service-ns"
-	})
-
-	Describe("NewLoadBalancer", func() {
-		Context("when everything is ok", func() {
-			var (
-				err     error
-				cfg     *rest.Config
-				testEnv *envtest.Environment
-			)
-
-			It("should not return error", func() {
-				testEnv = &envtest.Environment{}
-				cfg, err = testEnv.Start()
-				Expect(err).ShouldNot(HaveOccurred())
-
-				_, err = NewLoadBalancer(testClusterNameSpace, cfg, &testOwnerReference)
-				if err != nil {
-					klog.Infof("nicole, error is ", err)
-				} else {
-					klog.Infof("nicole, lb error is nothing")
-				}
-				Expect(err).ShouldNot(HaveOccurred())
-
-				err = testEnv.Stop()
-				Expect(err).ShouldNot(HaveOccurred())
-			})
-		})
-	})
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			cfg, err := testCase.testEnv.Start()
+			assert.NoError(t, err)
 
-	Describe("GetLoadBalancer", func() {
-		var (
-			err    error
-			exists bool
-		)
+			_, err = NewLoadBalancer(testClusterNameSpace, cfg, &testOwnerReference)
+			assert.NoError(t, err)
+			assert.Equal(t, testCase.err, err)
```

Comment:
This assert and the one above are checking the same thing I think. Can probably remove one of them.

Reply:
The upper one should be removed to be generic

Reply:
good catch, updated!